### PR TITLE
1284: Added a bunch of new outputs for naming module.

### PR DESCRIPTION
**Pull request by dsahoo-nexient**

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -1,9 +1,10 @@
-terraform 1.1.7
-tflint 0.34.1
-golang 1.17.6
-golangci-lint 1.46.2 
+conftest 0.44.1
+golang 1.20.8
+golangci-lint 1.53.3
 nodejs 16.13.2
```
Comment (chris-taylor-ntt) [inline .tool-versions:4]: You can remove nodejs and opa from this file; they should not be required for TF modules.

**Review by aaron-launch — APPROVED** [review]

File: main.tf [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
+}
+
 locals {
-  logical_product_name = trimspace(var.logical_product_name)
-  region               = var.use_azure_region_abbr ? lookup(local.azure_region_abbr_map, lower(trimspace(var.region)), trimspace(var.region)) : trimspace(var.region)
-  class_env            = trimspace(var.class_env)
-  cloud_resource_type  = trimspace(var.cloud_resource_type)
+  logical_product_family  = trimspace(var.logical_product_family)
+  logical_product_service = trimspace(var.logical_product_service)
+  region                  = var.use_azure_region_abbr ? lookup(local.azure_region_abbr_map, lower(trimspace(var.region)), trimspace(var.region)) : join("", split("-", var.region))
```
Comment (chris-taylor-ntt) [inline main.tf:11]: I think the final `var.region` here might also need a call to trimspace(), or you could end up with spaces in this value.

File: .tool-versions (hunk 1 above)
Comment (dsahoo-nexient) [reply .tool-versions:4]: done

File: main.tf (hunk 2 above)
Comment (dsahoo-nexient) [reply main.tf:11]: done

File: main.tf [hunk 3]
```diff
@@ -20,49 +28,66 @@ locals {
...
-    local.region,
-    local.class_env,
-    local.instance_env,
-    local.cloud_resource_type,
-  local.instance_resource) : local.standard
-
+  standard = lower(join(var.separator, local.variable_list))
+  # will remove any . or _ with a -. We don't want . as it will be parsed as unintended subdomains.
+  dns_compliant_standard         = replace(local.standard, "/[-_.]{1}/", "-")
+  minimal                        = lower(join(var.separator, [local.logical_product_family, local.logical_product_service, local.class_env, local.instance_env, local.cloud_resource_type]))
```
[the remark this answers is not in the document]
Comment (dsahoo-nexient) [reply main.tf:61]: done

File: main.tf [hunk 4]
```diff
@@ -20,49 +28,66 @@ locals {
...
+  minimal_without_any_separators = replace(local.minimal, "/[-_.]{1}/", "")
+  # Appends a 10 digit random number to the local.minimal and then trims it from the right until its under maximum_length limit.
+  minimal_random_suffix               = trimsuffix(substr(lower(join(var.separator, [local.logical_product_family, local.logical_product_service, random_integer.random_number.id])), 0, tonumber(var.maximum_length) - 1), var.separator)
```
[the remark this answers is not in the document]
Comment (dsahoo-nexient) [reply main.tf:64]: done

**Review by aaron-launch — APPROVED** [review]

**Review by bignastybryce — APPROVED** [review]

**Pull request merged**